### PR TITLE
Item management sup 10

### PR DESCRIPTION
**Pull request by FortuneAwaker**
Merging SUP-10 ticket with develop.

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/controller/ItemController.java [hunk 1]
```diff
@@ -1,13 +1,78 @@
...
+    }
+
+    @GetMapping(ApiConstants.URL_ACTIVE_PARAMETER)
+    public Page<ItemDto> getAllByActive(@PathVariable Boolean active,
+                                        @PageableDefault(size = 10) Pageable pageable) {
+        return itemService.findAllByActive(pageable, active);
+    }
+
+    @GetMapping(ApiConstants.URL_CATEGORY_PARAMETER)
+    public Page<ItemDto> getAll(@PathVariable String category, @PageableDefault(size = 10) Pageable pageable) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/ItemController.java]: what is default size in @PageableDefault?

File: src/main/java/by/itech/lab/supplier/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    private final CategoryService categoryService;
+
+    @PostMapping
+    public ResponseEntity<CategoryDto> save(@Valid @RequestBody CategoryDto categoryDto) {
+        return new ResponseEntity<>(categoryService.save(categoryDto), HttpStatus.CREATED);
+    }
+
+    @GetMapping(ApiConstants.URL_ACTIVE_PARAMETER)
+    public Page<CategoryDto> getAllByActive(@PathVariable Boolean active,
+                                            @PageableDefault(size = 10) Pageable pageable) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/CategoryController.java]: what is default size for @PageableDefault?

File: src/main/java/by/itech/lab/supplier/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+@AllArgsConstructor
+@Slf4j
+@RequestMapping(ApiConstants.URL_CATEGORY)
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @PostMapping
+    public ResponseEntity<CategoryDto> save(@Valid @RequestBody CategoryDto categoryDto) {
+        return new ResponseEntity<>(categoryService.save(categoryDto), HttpStatus.CREATED);
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/CategoryController.java]: Are there some special requirements for ResponseEntity?

File: src/main/java/by/itech/lab/supplier/controller/ItemController.java [hunk 4]
```diff
@@ -1,13 +1,78 @@
...
 @RequestMapping(URL_ITEM)
 public class ItemController {
+
+    private final ItemService itemService;
+
+    private final CategoryService categoryService;
+
+    @PostMapping
+    public ResponseEntity<ItemDto> save(@Valid @RequestBody ItemDto itemDto) {
+        return new ResponseEntity<>(itemService.save(itemDto), HttpStatus.CREATED);
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/ItemController.java]: Are there some special requirements for ResponseEntity?

File: src/main/java/by/itech/lab/supplier/domain/Item.java [hunk 5]
```diff
@@ -6,66 +6,37 @@
...
     @JoinColumn(name = "category_id")
-    public Category getCategory() {
-        return category;
-    }
-
-    public void setCategory(Category category) {
-        this.category = category;
-    }
+    private Category category;
+    @Column
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/domain/Item.java]: is this annotation necessary?

**Review by svareyko — CHANGES_REQUESTED** [review]

File: src/main/java/by/itech/lab/supplier/controller/CategoryController.java (hunk 2 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/controller/CategoryController.java]: done

File: src/main/java/by/itech/lab/supplier/controller/ItemController.java (hunk 1 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/controller/ItemController.java]: done

File: src/main/java/by/itech/lab/supplier/controller/ItemController.java (hunk 4 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/controller/ItemController.java]: done

File: src/main/java/by/itech/lab/supplier/controller/CategoryController.java (hunk 3 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/controller/CategoryController.java]: done

File: src/main/java/by/itech/lab/supplier/domain/Item.java (hunk 5 above)
Comment (FortuneAwaker) [reply src/main/java/by/itech/lab/supplier/domain/Item.java]: done

File: src/main/java/by/itech/lab/supplier/controller/ItemController.java [hunk 6]
```diff
@@ -1,13 +1,77 @@
...
+
+    @GetMapping(ApiConstants.URL_ACTIVE_PARAMETER)
+    public Page<ItemDto> getAllByActive(@PathVariable Boolean active,
+                                        @PageableDefault() Pageable pageable) {
+        return itemService.findAllByActive(pageable, active);
+    }
+
+    @GetMapping(ApiConstants.URL_CATEGORY_PARAMETER)
+    public Page<ItemDto> getAll(@PathVariable String category, @PageableDefault() Pageable pageable) {
+        CategoryDto found = categoryService.findByCategory(category);
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/ItemController.java]: this logic must be placed in the service layer

File: pom.xml [hunk 7]
```diff
@@ -63,6 +63,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>javax.validation</groupId>
```
Comment (svareyko) [inline pom.xml]: this dependency is already there

File: src/main/java/by/itech/lab/supplier/domain/Item.java (hunk 5 above)
Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/domain/Item.java]: its still here

**Review by svareyko — APPROVED** [review]

**Review by alexeyoverchenko — APPROVED** [review]

**Pull request merged**